lib/ibandit/iban.rb

```diff
@@ -383,6 +384,24 @@ def bsb_check?
       valid_modulus_check_branch_code?
     end
 
+    # rubocop:disable Metrics/AbcSize
+    def bank_code_passes_checksum_test?
+      return false if swift_bank_code.length != 9
+
+      code_digits = swift_bank_code.chars.map(&:to_i)
+      mod =
+        (
+          3 * (code_digits[0] + code_digits[3] + code_digits[6]) +
+          7 * (code_digits[1] + code_digits[4] + code_digits[7]) +
+          1 * (code_digits[2] + code_digits[5] + code_digits[8])
```

Comment:
Given we have this snippet of code in both banking_integrations for the Geo bank account stuff I'm trying to understand why we need to duplicate it in two places?

Reply:
My understanding is that the geo bank account flow is independent and doesn't share any code with the old version of bank accounts (pseudo ibans). Until we decide to move completely to geo bank accounts we want to support both versions (we currently persist and send both of them down to Bank ops).